modules/images/dominant-color/load.php

```diff
@@ -0,0 +1,336 @@
+}
+add_filter( 'wp_enqueue_scripts', 'dominant_color_add_inline_style' );
+
+
+/**
+ * Overloads wp_image_editors() to load the extended classes.
+ *
+ * @since n.e.x.t
+ *
+ * @return string[] Registered image editors class names.
+ */
+function dominant_color_set_image_editors() {
+	if ( ! class_exists( 'Dominant_Color_Image_Editor_GD' ) ) {
+		require_once __DIR__ . '/class-dominant-color-image-editor-gd.php';
+	}
+	if ( ! class_exists( 'Dominant_Color_Image_Editor_Imagick' ) ) {
+		require_once __DIR__ . '/class-dominant-color-image-editor-imagick.php';
+	}
+
+	return array( 'Dominant_Color_Image_Editor_GD', 'Dominant_Color_Image_Editor_Imagick' );
+}
+
+/**
+ * Get dominant color of image
+ *
+ * @since n.e.x.t
+ *
+ * @param int $attachment_id the image id.
+ * @return string|WP_Error the dominant color of the image or WP_Error on error.
+ */
+function dominant_color_get_dominant_color( $attachment_id ) {
+	$file = wp_get_attachment_file_size( $attachment_id );
+	if ( ! $file ) {
+		$file = get_attached_file( $attachment_id );
+	}
+	add_filter( 'wp_image_editors', 'dominant_color_set_image_editors' );
+	$editor = wp_get_image_editor( $file );
+	remove_filter( 'wp_image_editors', 'dominant_color_set_image_editors' );
+
+	if ( is_wp_error( $editor ) ) {
+		return $editor;
+	}
+	if ( ! method_exists( $editor, 'get_dominant_color' ) ) {
+		return new WP_Error( 'unable_to_find_method', __( 'Unable to find get_dominant_color method', 'performance-lab' ) );
+	}
+
+	$dominant_color = $editor->get_dominant_color();
+	if ( is_wp_error( $dominant_color ) ) {
+		return $dominant_color;
+	}
+
+	return $dominant_color;
+}
+
+/**
+ * Works out if color has transparency
+ *
+ * @since n.e.x.t
+ *
+ * @param int $attachment_id the attachment id.
+ * @return bool|WP_Error True if the color has transparency or WP_Error on error.
+ */
+function dominant_color_has_transparency( $attachment_id ) {
+	$file = wp_get_attachment_file_size( $attachment_id );
+	if ( ! $file ) {
+		$file = get_attached_file( $attachment_id );
+	}
+	add_filter( 'wp_image_editors', 'dominant_color_set_image_editors' );
+	$editor = wp_get_image_editor( $file );
+	remove_filter( 'wp_image_editors', 'dominant_color_set_image_editors' );
+
+	if ( is_wp_error( $editor ) ) {
+		return $editor;
+	}
+	if ( ! method_exists( $editor, 'has_transparency' ) ) {
+		return new WP_Error( 'unable_to_find_method', __( 'Unable to find has_transparency method', 'performance-lab' ) );
+	}
+	$has_transparency = $editor->has_transparency();
+	if ( is_wp_error( $has_transparency ) ) {
+		return $has_transparency;
+	}
+
+	return $has_transparency;
+}
+
+/**
+ * Get file path of image based on size.
+ *
+ * @since n.e.x.t
+ *
+ * @param int    $attachment_id Attachment ID for image.
+ * @param string $size          Optional. Image size. Default 'thumbnail'.
+ *
+ * @return false|string Path to an image or false if not found.
+ */
+function wp_get_attachment_file_size( $attachment_id, $size = 'medium' ) {
```

Comment:
See previous comment:
```suggestion
function wp_get_attachment_file_path( $attachment_id, $size = 'medium' ) {
```